File: ariltodolist/lib/models/task.dart
class Task {
  final int? id;
  final String title;
  final String priority;
  final String dueDate;
  final bool isDone;

  Task({
    this.id,
    required this.title,
    required this.priority,
    required this.dueDate,
    this.isDone = false,
  });

  factory Task.fromJson(Map<String, dynamic> json) {
    return Task(
      id: json['id'],
      title: json['title'],
      priority: json['priority'],
      dueDate: json['due_date'],
      isDone: json['is_done'].toString() == '1' || json['is_done'] == true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "title": title,
      "priority": priority,
      "due_date": dueDate,
      "is_done": isDone ? 1 : 0,
    };
  }
}

File: arilhatodo/app/Models/Task.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Task extends Model
{
    use HasFactory;

    protected $fillable = [
        'title',
        'priority',
        'due_date',
        'is_done',
        'user_id',
    ];

    // Relasi Task → User
    public function user()
    {
        return $this->belongsTo(User::class);
    }
}
